src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -1523,4 +1528,40 @@ private function getCustomerMessageErrorMapping(Exception $exception): array
             ],
         ];
     }
+
+    /**
+     * @param int $orderId
+     *
+     * @return int|null
+     */
+    private function getPreviousOrderId(int $orderId)
+    {
+        try {
+            /** @var OrderForViewing $previousOrderForViewing */
+            $previousOrderForViewing = $this->getQueryBus()->handle(new GetOrderForViewing($orderId - 1));
```

Comment:
We cannot use `$orderId - 1)` that's the tricky part of this feature 😄 
- if someone manually deletes a row from mysql it does not work
- if mysql jumps one increment (can happen) it does not work

We need to retrieve the real `previous MySQL ID` which is __probably__ "$orderId - 1" but we need to be sure.